client/my-sites/checkout/composite-checkout/hooks/has-renewal.ts

```diff
@@ -13,5 +13,5 @@ export function isRenewalInLineItems( items ) {
 }
 
 export function isLineItemARenewal( item ) {
-	return 'renewal' === ( item.wpcom_meta?.extra?.purchaseType ?? '' );
+	return 'renewal' === ( ( item.wpcom_meta ?? item )?.extra?.purchaseType ?? '' );
```

Comment:
It's a bit awkward that we have two types of cart items (my fault, I suppose, for introducing the standardized one for the composite-checkout package, and not making sure to always use it) but I'd rather not mix the functions that use those types. (Side note: I'm surprised that this function has no `WPCOMCartItem` type on `item` since it's a TS file.)

I suggest that instead of modifying this function, we just use the existing `isRenewal()` from `client/lib/cart-values/cart-items.js` if the item is a `ResponseCartProduct`.

Reply:
Typed `extra` and limited `isRenewal` to just expecting the `extra` object property.

Reply:
Sounds good! I think you can remove this change, in that case.